src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -1338,6 +1367,15 @@ class TrainDialogs extends React.Component<Props, ComponentState> {
                             text={Util.formatMessageId(intl, FM.TRAINDIALOGS_TREEVIEW_BUTTON)}
                         />
                     }
+                    <OF.DefaultButton
```

Comment:
I'd disable or hide this button when it's not needed?   It's kinda confusing.   Maybe a warning when you click on it?  User is tarting a big operation and may have no idea what they are about to do

Reply:
Yes, both of these were in the description as well.  We need to be able to determine when it's not needed so it's related to that question.

And yes, what do you consider a big operation 2 or more dialogs? or any replay all? or 10 dialogs?

Reply:
If we switch to "replay selected" I don't think disable makes sense.   An alternative to the warning would be a progress modal with a cancel button that lets user quit in the middle.  I'm using that for the testing stuff

Reply:
See [TestWaitModal.tsx](https://github.com/microsoft/ConversationLearner-UI/blob/master/src/components/modals/TestWaitModal.tsx)

Reply:
Any thoughts on the above?  Think that would help with folks not knowing they may be starting something that could take 20 minutes?   Something like this but "Replaying" instead of "Testing"

![image](https://user-images.githubusercontent.com/6528486/61826316-04d41700-ae17-11e9-928e-ccfb5a4a0237.png)

Reply:
I didn't realize this could take 20 minutes, only thought it was a lot of network calls. I suppose we don't know how long their Code callbacks will take and they could take very long time.

Anyways, will try to add this.

Reply:
Yeah, testing can take a while at it has to replay every test dialog through very step (entity extraction, scorer step, etc) and you may be testing 100+ .transcript files

Reply:
Hmm, looks like the larger spinner blocks interaction with the modal anyways. I'm guessing since the testing functions were new they didn't have dependent logic to work around.

I see two options:
1. remove the modal and go with as 
2. add some other flag to the action to not enable the spinner during this special type of replay
Seems like we keep adding these flags for special conditions which adds a lot of complexity. Seems like we should remove the global at some point if we're going to combine it with these local modal spinner.

Reply:
Well went ahead with #2.